test/unit-tests/components/viewmodels/right_panel/RoomSummaryCardTopicViewModel-test.tsx

```diff
@@ -0,0 +1,89 @@
+/*
+Copyright 2025 New Vector Ltd.
+SPDX-License-Identifier: AGPL-3.0-only OR GPL-3.0-only
+Please see LICENSE files in the repository root for full details.
+*/
+
+import { renderHook } from "jest-matrix-react";
+import { act, type SyntheticEvent } from "react";
+
+import { useRoomTopicViewModel } from "../../../../../src/components/viewmodels/right_panel/RoomSummaryCardTopicViewModel";
+import { mkStubRoom, stubClient } from "../../../../test-utils";
+import defaultDispatcher from "../../../../../src/dispatcher/dispatcher";
+import { onRoomTopicLinkClick } from "../../../../../src/components/views/elements/RoomTopic";
+
+jest.mock("../../../../../src/components/views/elements/RoomTopic");
+
+describe("RoomSummaryCardTopicViewModel", () => {
+    const client = stubClient();
+    const mockRoom = mkStubRoom("!room:example.com", "Test Room", client);
+    const mockUserId = "@user:example.com";
+    const mockEvent = { preventDefault: jest.fn(), stopPropagation: jest.fn() } as unknown as SyntheticEvent;
+
+    beforeEach(() => {
+        // Mock room client's getSafeUserId
+        mockRoom.client.getSafeUserId = jest.fn().mockReturnValue(mockUserId);
+        jest.spyOn(defaultDispatcher, "dispatch");
+
+        (onRoomTopicLinkClick as jest.Mock).mockReset();
+    });
+
+    afterEach(() => {
+        jest.restoreAllMocks();
+    });
+
+    function render() {
+        return renderHook(() => useRoomTopicViewModel(mockRoom));
+    }
+
+    it("should initialize with expanded state", () => {
+        const { result } = render();
+        expect(result.current.expanded).toBe(true);
+    });
+
+    it("should toggle expanded state on click", () => {
+        const { result } = render();
+
+        act(() => {
+            result.current.onExpandedClick(mockEvent);
+        });
+
+        expect(result.current.expanded).toBe(false);
+    });
+
+    it("should handle edit click", () => {
+        const { result } = render();
+        result.current.onEditClick(mockEvent);
+        expect(defaultDispatcher.dispatch).toHaveBeenCalledWith({ action: "open_room_settings" });
+    });
+
+    it("should handle topic link clicks when the target is an anchor element", () => {
+        const { result } = render();
+        const mockAnchorEvent = { target: document.createElement("a") };
```

Comment:
```suggestion
        const mockAnchorEvent =  new HTMLAnchorElement();
        mockAnchorEvent.target: document.createElement("a");
```

Then you don't need to cast it to any